tests/test_axis.py

```diff
@@ -597,6 +597,30 @@ class TestCirculationMonitor(Axis360Test):
         last_checked=datetime.datetime(2015, 5, 20, 2, 9, 8),
     )
 
+    def test_run(self):
+        class Mock(Axis360CirculationMonitor):
+            def catch_up_from(self, start, cutoff, progress):
+                self.called_with = (start, cutoff, progress)
+        monitor = Mock(self._db, self.collection, api_class=MockAxis360API)
+
+        # The first time run() is called, catch_up_from() is asked to
+        # find events between DEFAULT_START_TIME and the current time.
+        monitor.run()
+        start, cutoff, progress = monitor.called_with
+        now = datetime.datetime.utcnow()
+        eq_(monitor.DEFAULT_START_TIME, start)
+        assert (now - cutoff).total_seconds() < 2
+
+        # The second time run() is called, catch_up_from() is asked
+        # to find events between five minutes before the last cutoff,
+        # and what is now the current time.
+        monitor.run()
+        new_start, new_cutoff, new_progress = monitor.called_with
+        now = datetime.datetime.utcnow()
+        before_old_cutoff = cutoff - monitor.OVERLAP
+        eq_(before_old_cutoff, new_start)
+        assert (now - new_cutoff).total_seconds() < 2
+
```

Comment:
I added end-to-end tests of `run` or `run_once` to a lot of the monitors I changed, even though the code is technically already tested as part of the `TimelineMonitor` test coverage in core.

First, I'm concerned that unless there's an end-to-end test, future changes to the `Monitor` API will break individual monitors without breaking the tests.

Second, my next branch is going to add status reporting to all of these monitors, and a test like this is often the simplest place to put the code that checks whatever status was reported.